rest-server/core/src/main/scala/com/actionml/core/backup/FSMirroring.scala

```diff
@@ -18,70 +18,60 @@
 package com.actionml.core.backup
 
 import java.io.{File, FileWriter, PrintWriter}
-import java.nio.file.{Files, Path, Paths}
 
+import cats.data.Validated
+import cats.data.Validated.{Invalid, Valid}
 import com.actionml.core.template.Engine
+import com.actionml.core.validate.{ValidRequestExecutionError, ValidateError}
 
 import scala.io.Source
 
 /**
   * Mirroring implementation for local FS.
   */
 
-// TODO: not using injection so need a trait
-//object FSMirroring extends Mirroring {
-trait FSMirroring extends Mirroring {
+class FSMirroring(mirrorContainer: String) extends Mirroring(mirrorContainer) {
 
```

Comment:
I don't believe we can set the `mirrorContainer` at construction time since it is configured the `engine.json` file which is subject to update while the server is running.